coil/src/androidTest/java/com/google/accompanist/coil/CoilTest.kt

```diff
@@ -594,22 +384,11 @@ class CoilTest {
     @Test(expected = IllegalArgumentException::class)
     fun data_painter_throws() {
         composeTestRule.setContent {
-            CoilImage(
-                data = ColorPainter(Color.Magenta),
+            AsyncImage(
+                state = rememberCoilAsyncImageState(ColorPainter(Color.Magenta)),
                 contentDescription = null,
                 modifier = Modifier.size(128.dp, 128.dp),
             )
         }
     }
 }
```

Comment:
Should we test hoisting the state? Should you be able to pass this to multiple `AsyncImage`s?

Reply:
I'm not sure on this one, and I'm not even sure this is something I want to support. Hoisting state here is more for being able observe the 'internal' state, rather than having multiple `AsyncImage`s driven from the same state.

We already have tests for that, with the new `loading` content tests, which use the state's `loadState` to drive a `Crossfade`. WDYT?

Reply:
Yeah that's fair and saw those tests after adding this comment!!